internal/grpcutil/compressor.go

```diff
@@ -45,3 +46,41 @@ func RegisteredCompressors() string {
 	}
 	return strings.Join(RegisteredCompressorNames, ",")
 }
+
+// ValidateSendCompressor returns an error when given compressor name cannot be
+// handled by the server or the client based on the advertised compressors.
+func ValidateSendCompressor(name, clientAdvertisedCompressors string) error {
+	if name == "identity" {
+		return nil
+	}
+
+	if !IsCompressorNameRegistered(name) {
+		return fmt.Errorf("compressor not registered: %s", name)
+	}
+
+	if !compressorExists(name, clientAdvertisedCompressors) {
+		return fmt.Errorf("client does not support compressor: %s", name)
+	}
+
+	return nil
+}
+
+// compressorExists returns true when the given name exists in the comma
+// separated compressor list.
+func compressorExists(name, compressors string) bool {
+	var (
+		i      = 0
+		length = len(compressors)
+	)
+	for j := 0; j <= length; j++ {
+		if j < length && compressors[j] != ',' {
```

Comment:
Use `strings.Split` instead to get a slice of compressor names from `compressors` and then iterate through the slice to see if `name` exists?

Reply:
Done, I skipped that earlier to avoid split string allocation